chminfo_service/__init__.py

```diff
@@ -0,0 +1,241 @@
+import os
+import re
+import hashlib
+import tempfile
+import HTMLParser
+import logging
+from chm import chm
+
+from crits.services.core import Service, ServiceConfigError
+
+logger = logging.getLogger(__name__)
+
+class CHMInfoService(Service):
+    """
+    Microsoft Compiled HTML Help file information service
+    - Extract information from CHM
+    - Provde details about suspicious behaviour within a CHM
+    """
+    name = "chminfo"
+    version = '1.0.0'
+    supported_types = ['Sample']
+    description = "Generate information about Windows CHM files."
+    added_files = []
+
+    chmparse = chm.CHMFile()
+    item_string = {r'x-oleobject':'CHM contains reference to OLE Object.',
+                    r'<script':'CHM contains JavaScript.',
+                    r'.savetofile':'CHM contains a function to save data to file.',
+                    r'document.write(':'CHM contains a function to save data to file.',
+                    r'adodb.stream':'CHM creates ADO steam object for file access.',
+                    r'msxml2.xmlhttp':'CHM uses an XHLHTTP object to create a network connection.',
+                    r'system.net.webclient':'CHM uses the PowerShell WebClient class to create a network connection.',
+                    r'cmd.exe':'CHM references Windows Command Prompt (cmd).',
+                    r'cscript':'CHM references Console Based Script Host (cscript).',
+                    r'wscript':'CHM references Windows Based Script Host (wscript).',
+                    r'rundll32':'CHM references Windows host process (rundll32).',
+                    r'powershell':'CHM references Windows PowerShell.',
+                    r'end if':'CHM contains \'if\' statement.',
+                    }
+    item_regex = {r'<iframe\s.*src="([^\"]*)".*>':'CHM file creates an IFRAME',
+                    r'<object\s[^>]+codebase=\"([^\"]*)\"':'CHM contains object that references external code',
+                    r'<object\s[^>]+codebase=\'([^\"]*)\'':'CHM contains object that references external code',
+                    r'<object\s[^>]+data=\"([^\"]*)\"':'CHM contains object that references external code',
+                    r'<object\s[^>]+data=\'([^\"]*)\'':'CHM contains object that references external code',
+                    r'createobject\(([^\)]*)':'CHM attempts to create an object',
+                    r'.downloadfile\(([^\)^,]*)': 'CHM attempts to download a file',
+                    r'.exec\(([^\)]*)':'CHM attempts to execute a file',
+                    r'.shellexecute\(([^\)]*)': 'CHM attempts to execute a file',
+                }
+
+    @staticmethod
+    def valid_for(obj):
+        chm_magic = '\x49\x54\x53\x46\x03\x00\x00\x00\x60\x00\x00\x00'
```

Comment:
A quick look at the wikipedia page says the magic should be "ITSF". I have no clue what the other bytes on the end of your string are, but I'm concerned they may be too restrictive. Any thoughts on this?